applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/formfinding_strategy.hpp

```diff
@@ -0,0 +1,265 @@
+#include "custom_utilities/project_vector_on_surface_utility.h"
+#include "includes/model_part_io.h"
+#include "includes/kratos_filesystem.h"
+#include "input_output/vtk_output.h"
+#include "includes/gid_io.h"
+#include "structural_mechanics_application_variables.h"
+
+namespace Kratos
+{
+    template<class TSparseSpace,
+    class TDenseSpace, // = DenseSpace<double>,
+    class TLinearSolver //= LinearSolver<TSparseSpace,TDenseSpace>
+    >
+    class FormfindingStrategy
+        : public ResidualBasedNewtonRaphsonStrategy<TSparseSpace, TDenseSpace, TLinearSolver>
+    {
+        public:
+            ///@name Type Definitions
+            ///@{
+            typedef ConvergenceCriteria<TSparseSpace, TDenseSpace> TConvergenceCriteriaType;
+
+            // Counted pointer of ClassName
+            KRATOS_CLASS_POINTER_DEFINITION(FormfindingStrategy);
+
+            typedef ResidualBasedNewtonRaphsonStrategy<TSparseSpace, TDenseSpace, TLinearSolver> BaseType;
+            typedef typename BaseType::TBuilderAndSolverType TBuilderAndSolverType;
+            typedef typename BaseType::TSchemeType TSchemeType;
+            typedef typename BaseType::TSystemMatrixType TSystemMatrixType;
+            typedef typename BaseType::TSystemVectorType TSystemVectorType;
+            typedef GidIO<> IterationIOType;
+            typedef IterationIOType::Pointer IterationIOPointerType;
+
+            ///@}
+            ///@name Life Cycle
+
+            ///@{
+
+            /**
+            * Constructor.
+            */
+
+            FormfindingStrategy(
+                ModelPart& model_part,
+                typename TSchemeType::Pointer pScheme,
+                typename TLinearSolver::Pointer pNewLinearSolver,
+                typename TConvergenceCriteriaType::Pointer pNewConvergenceCriteria,
+                ModelPart& rFormFindingModelPart,
+                bool WriteNewReferenceFile,
+                std::string PrintingFormat,
+                Parameters ProjectionSetting,
+                int MaxIterations = 30,
+                bool CalculateReactions = false,
+                bool ReformDofSetAtEachStep = false,
+                bool MoveMeshFlag = false
+                )
+                : ResidualBasedNewtonRaphsonStrategy<TSparseSpace, TDenseSpace, TLinearSolver>(model_part, pScheme,
+                    pNewLinearSolver,
+                    pNewConvergenceCriteria,
+                    MaxIterations,
+                    CalculateReactions,
+                    ReformDofSetAtEachStep,
+                    MoveMeshFlag),
+                    mProjectionSettings(ProjectionSetting),
+                    mrFormFindingModelPart(rFormFindingModelPart),
+                    mPrintingFormat(PrintingFormat),
+                    mWriteNewReferenceFile(WriteNewReferenceFile)
+                 {
+                    InitializeIterationIO();
+                 }
+
+            // constructor with Builder and Solver
+            FormfindingStrategy(
```

Comment:
do you need both constructors?
I suggest to only keep the one that you actually need.

If you need both then please check if you can call one constructor from the other like it is done with `ModelPart` to reduce code duplication